### DISCUSSION TITLE
text() with Special characters

### DISCUSSION THREAD

**Discussion by rebornfz** [libvips/libvips#2227] [Q&A] [answered]
I encountered special characters “&” when rendering text with text(). It went wrong and threw
‘text: invalid markup in text’

How should I render these special characters?

**Comment by rebornfz** [comment]
"<"  also went wrong, but ">" worked will.

**Comment by jcupitt** [comment, marked as answer]
You can put markup in strings, for example:

```
txt = pyvips.Image.text("Hello <i>World!</i>")
```

So you have to escape special characters, just as you would with HTML. You can use eg. `&lt;` for `<`, just like HTML.

See: 

https://developer.gnome.org/pygtk/stable/pango-markup-language.html

**Comment by rebornfz** [comment]
I can render most special characters, except "&" and "<".
For example,
text_image = vips::VImage::text("~`!@#$%^*()_+-=,.?/>") //worked well
text_image = vips::VImage::text("&") // went wrong
text_image = vips::VImage::text("<") //went wrong
text_image = vips::VImage::text("<i>&</i>") //went wrong
text_image = vips::VImage::text("<i>&lt</i>") //went wrong
It's so strange
Puting markup in strings seems not work.

**Reply by jcupitt** [reply]
You need the `;` as well. Try `"&lt;"`.

**Reply by rebornfz** [reply]
Thank you very much! It worked well.
